# Obstacle feed mode (0 = sensor mode, 1 = OptiTrack mode, 2 = predefined obstacles, 4 = Pedsim)
# need to recompile after changing!
obstacle_feed_mode: 4

# Debugging info
activate_debug_output: false
activate_visualization: true
activate_timing_output: true

# Rate at which ellipse obstacles should be send out
update_rate: 20
discretization_steps: 15
prediction_horizon: 3

# Define the coordinate frames
frames:
    planning_frame: map             # For predefined obstacles
    robot_frame: odom          # For actual measured obstacles
kalman: false
publish:
    obstacles:  ellipse_objects_feed #/nn_jackal/clusters
    obstacles_vis: visualize_ellipses_array
    pub_people: /people
subscribe:
    detections: none #detection_3d
    optitrack: none #future_obj_track
    pedestrians: none #/spencer/perception/tracked_persons_confirmed_by_yolo
    prius: none #/future_obj_track
    pedsim: /pedsim_visualizer/tracked_persons

# Detection criteria
detected_obstacles:
    distance_threshold: 50
    obstacle_number: 8
    min_obstacle_volume: 0.03
    max_obstacle_volume: 1.0
    obstacle_size: 0.3

# Create predefined obstacles
predefined_obstacles:
    pose_x: [85, 55, 10]
    pose_y: [0, -1, -2]
    heading: [0, 0, 0]
    v_x: [-0.5, 0.15, 0.3]
    v_y: [0.0, 0.0, 0.0]
    dimensions:
        minor_semiaxis: [0.3, 0.3, 0.3]
        major_semiaxis: [0.3, 0.3, 0.3]
